includes/woocommerce-memberships/class-admin.php

```diff
@@ -0,0 +1,149 @@
+			__( 'Newspack Network', 'newspack-network' ),
+			array( __CLASS__, 'render_meta_box' ),
+			self::MEMBERSHIPS_CPT,
+			'advanced'
+		);
+	}
+
+	/**
+	 * Renders the meta box.
+	 *
+	 * @param \WP_Post $post The post object.
+	 */
+	public static function render_meta_box( $post ) {
+		$network_id = get_post_meta( $post->ID, self::NETWORK_ID_META_KEY, true );
+
+		wp_nonce_field( 'newspack_network_save_membership_plan', 'newspack_network_save_membership_plan_nonce' );
+		?>
+		<label for="newspack-network-id"><?php esc_html_e( 'Network ID', 'newspack-network' ); ?></label>
+		<input type="text" id="newspack-network-id" name="newspack_network_id" value="<?php echo esc_attr( $network_id ); ?>" />
+		<p><?php esc_html_e( 'If a Network ID is informed, the user memberships will be propagated to the network. The user will be granted with the membership with the matching Network ID in all other sites in the network.', 'newspack-network' ); ?></p>
+		<?php
+	}
+
+	/**
+	 * Saves the meta box.
+	 *
+	 * @param int $post_id The post ID.
+	 */
+	public static function save_meta_box( $post_id ) {
+
+		$post_type = sanitize_text_field( $_POST['post_type'] ?? '' );
+
+		if ( self::MEMBERSHIPS_CPT !== $post_type ) {
+			return;
+		}
+
+		if ( ! isset( $_POST['newspack_network_save_membership_plan_nonce'] ) ||
+			! wp_verify_nonce( sanitize_text_field( $_POST['newspack_network_save_membership_plan_nonce'] ), 'newspack_network_save_membership_plan' )
+		) {
+			return;
+		}
+
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+			return;
+		}
+
+		$post_type_object = get_post_type_object( $post_type );
+
+		if ( ! current_user_can( $post_type_object->cap->edit_post, $post_id ) ) {
+			return;
+		}
+
+		$network_id = sanitize_text_field( wp_unslash( $_POST['newspack_network_id'] ?? '' ) );
+		update_post_meta( $post_id, self::NETWORK_ID_META_KEY, $network_id );
+	}
+
+	/**
+	 * Filters the edit post link for user memberships managed in another site of the network.
+	 *
+	 * @param string $link The edit post link.
+	 * @param int    $post_id The post ID.
+	 * @return string
+	 */
+	public static function get_edit_post_link( $link, $post_id ) {
+		$managed_by_newspack_network = get_post_meta( $post_id, self::NETWORK_MANAGED_META_KEY, true );
+		if ( $managed_by_newspack_network ) {
+			$link = str_replace( get_bloginfo( 'url' ), $managed_by_newspack_network, $link );
+		}
+		return $link;
+	}
+
+	/**
+	 * Filters the row actions for user memberships managed in another site of the network.
+	 *
+	 * @param string[] $actions An array of row action links. Defaults are
+	 *                          'Edit', 'Quick Edit', 'Restore', 'Trash',
+	 *                          'Delete Permanently', 'Preview', and 'View'.
+	 * @param WP_Post  $post    The post object.
+	 * @return array
+	 */
+	public static function post_row_actions( $actions, $post ) {
+		$managed_by_newspack_network = get_post_meta( $post->ID, self::NETWORK_MANAGED_META_KEY, true );
```

Comment:
This will also be true on the membership-originating site, resulting in a confusing "Managed in <this-site>" message.